tensorboard/backend/application.py

```diff
@@ -451,14 +456,33 @@ def _serve_plugins_listing(self, request):
           A werkzeug.Response object.
         """
         response = collections.OrderedDict()
+        eid = plugin_util.experiment_id(request.environ)
+        plugins_with_data = (
+            self._data_provider.list_active_plugins(eid)
+            if self._data_provider is not None
+            else None
```

Comment:
Simpler below if this just defaults to the empty set?  Then we don't need a branch.

Reply:
Out of laziness I probably would have had list_plugins() default to frozenset(), but it's true that it is possibly useful down the road to be able to distinguish the two cases so I'm fine leaving it as you have it here.